### PR TITLE
Implement TunInterfaceOptions

### PR DESCRIPTION
**Pull request by davnotdev**
Implementation for #113.

This compiles on MacOS and Linux, but I couldn't cross compile for Windows, so I hope the tests will reveal if there are any errors. Another thing is with the naming of the options. For example, I stuck with `no_pi`, but I have a feeling that `no_packet_info` or `linux_no_pi` would be better names.

### CODE REVIEW

File: tun/src/options.rs [hunk 1]
```diff
@@ -0,0 +1,38 @@
+use fehler::throws;
+use std::io::Error;
+
+use super::TunInterface;
+
+#[derive(Default)]
+pub struct TunInterfaceOptions {
```
Comment (conradev) [inline tun/src/options.rs:7]: How does `TunOptions` sound? Just a bit shorter, I think the interface part could be inferred

File: tun/src/windows/mod.rs [hunk 2]
```diff
@@ -15,15 +17,22 @@ pub struct TunInterface {
 impl TunInterface {
     #[throws]
     pub fn new() -> TunInterface {
-        let name = U16CString::from(u16cstr!("Burrow"));
+        Self::new_with_options(TunInterfaceOptions::new())?
+    }
+
+    #[throws]
+    pub(crate) fn new_with_options(options: TunInterfaceOptions) -> TunInterface {
+        let name_owned = options.name.unwrap_or("Burrow".to_owned());
```
Comment (conradev) [inline tun/src/windows/mod.rs:25]: This seems good for now, but I am going to investigate what happens if we pass an empty name. It appears as if `wintun` might have handling for this

**Review by conradev — APPROVED** [review]
These changes look great! Other than perhaps changing the name, it's ready to merge

File: tun/src/options.rs (hunk 1 above)
Comment (davnotdev) [reply tun/src/options.rs]: Yep, `TunOptions` sounds more concise.

**Pull request merged**